nixos/modules/services/misc/weechat.nix

```diff
@@ -0,0 +1,103 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.weechat;
+in
+
+{
+  options.services.weechat = {
+    enable = mkEnableOption "weechat";
+    init = mkOption {
+      description = "Weechat commands applied at start, one command per line.";
+      example = ''
+        /set relay.network.password correct-horse-battery-staple
+        /relay add weechat 9001
+      '';
+      type = types.str;
+      default = "";
+    };
+    root = mkOption {
+      description = "Weechat state directory.";
+      type = types.str;
+      default = "/var/lib/weechat";
+    };
+    scripts = mkOption {
+      description = "Custom scripts to be loaded by the weechat module.";
+      type = with types; listOf (submodule {
+        options = {
+          name = mkOption {
+            type = str;
+            description = "Path to the script in the derivation.";
+          };
+          script = mkOption {
+            type = types.package;
+            description = "Package which contains the actual script.";
+          };
+        };
+      });
+      default = [];
+      example = literalExample ''
+        [
+          { name = "share/jabber.py"; script = pkgs.weechat-xmpp }
```

Comment:
Why not just `"${pkgs.weechat-xmpp}/share/jabber.py"`? Or maybe even force the script packages to be in a standard format and only require the package.

Reply:
you're absolutely right! I originally wanted to define a pre-defined output structure for derivations which feature weechat scripts, but dropped the idea later and kept the (admittedly too complex submodule type).